src/shared/awsContext.ts

```diff
@@ -41,6 +44,7 @@ export class DefaultAwsContext implements AwsContext {
     public readonly onDidChangeContext: vscode.Event<ContextChangeEventsArgs>
     private readonly _onDidChangeContext: vscode.EventEmitter<ContextChangeEventsArgs>
     private shim?: CredentialsShim
+    public lastTouchedRegion: string
```

Comment:
should `undefined` be used to model "not set" ?
```suggestion
    public lastTouchedRegion: string | undefined
```

Reply:
I originally had this and the changed it to have the string 'None' when it was not set. In retrospect, I can't remember why I ever swapped and I think you're correct in that defaulting to undefined makes a lot more sense.